lib/commands/push.ts

```diff
@@ -374,6 +374,21 @@ export default class PushCmd extends Command {
 			organization: [Organization];
 		};
 
+		// * Temporary fix *
+		// When doing `$expand organization` on a public app (when not the owner)
+		// a partial document is returned without any organization data.
+		// So e.g. `balena push balena-sound` will break this commands logic.
+		// The balena cloud builder api will soon change to accept application slugs
+		// instead of `owner, app`, after which we will not need to do the expand
+		//
+		// Users should not be pushing to public apps anyway, so for now catch this situation:
+		if (!application.organization[0]) {
+			throw new ExpectedError(stripIndent`
+			You do no have rights to push to the application ${application.app_name}.
```

Comment:
* Would it be better that this was `application.slug` instead of `application.app_name`? (I did not test, needs testing...) The slug, assuming it contains org / user account details, might help clarify to the user that the push was attempted to someone else's application.
* no → not
* Is the word "permissions" more common than "rights"? -  
`You do not have permissions to push to application ${application.app_name}.`

Reply:
> Would it be better that this was application.slug instead of application.app_name

In these cases the partial document you get back from the API doesn't contain the slug.  (Though may be a case of changing the query).

> no → not
:-1:

> Is the word "permissions" more common than "rights"? -
Yeah probably :+1:

Reply:
> [...] the partial document you get back from the API doesn't contain the slug. Though may be a case of changing the query.

OK. If a small query change could reveal the slug, I think it would be worth it. Otherwise, something like (for example) _"You do not have permissions to push to application balena-sound"_ could be misleading to a user who had a `balena-sound` application in some of their orgs. Although, if this error message was "impossible" in this situation, then it would not be misleading...